.github/workflows/unit-and-integration-tests.yaml

```diff
@@ -0,0 +1,169 @@
+name: Intel Device Plugins K8s Operator Unit and Integration Tests
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  call-inclusive-naming-check:
+    name: Inclusive naming
+    uses: canonical/inclusive-naming/.github/workflows/woke.yaml@main
+    with:
+      fail-on-error: "true"
+  
+  lint-and-unit-tests:
+    name: Lint and unit tests
+    uses: charmed-kubernetes/workflows/.github/workflows/lint-unit.yaml@main
+    with:
+      python: "['3.10', '3.11']"
+    needs:
+      - call-inclusive-naming-check
+
+  integration-tests:
+    name: Integration tests on Testflinger instance with Intel GPU
+    runs-on: [self-hosted, testflinger]
+    steps:
+      - name: Submit job to testflinger
+        uses: canonical/testflinger/.github/actions/submit@main
+        with:
+          poll: true
+          job: |
```

Comment:
Can we use `job-path` and provide a job file instead of providing the raw contents ? i think that if we use a file, it can be re-used by for different needs ; see https://github.com/canonical/testflinger/blob/8bd2a0f277556fe4b7524155837910dd6fed402c/.github/actions/submit/action.yaml#L7

Reply:
Let me give it a try

Reply:
Done